src/main/java/teammates/ui/webapi/action/LegacyUrlMapper.java

```diff
@@ -31,10 +33,17 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOExc
         switch (uri) {
         case Const.LegacyURIs.INSTRUCTOR_COURSE_JOIN:
             key = req.getParameter(Const.ParamsNames.REGKEY);
-            redirectUrl = Config.getFrontEndAppUrl(Const.WebPageURIs.JOIN_PAGE)
+            String institute = req.getParameter(Const.ParamsNames.INSTRUCTOR_INSTITUTION);
+            AppUrl newUrl = Config.getFrontEndAppUrl(Const.WebPageURIs.JOIN_PAGE)
                     .withRegistrationKey(key)
-                    .withParam(Const.ParamsNames.ENTITY_TYPE, Const.EntityType.INSTRUCTOR)
-                    .toString();
+                    .withParam(Const.ParamsNames.ENTITY_TYPE, Const.EntityType.INSTRUCTOR);
+            // TODO: Remove the following code block for mac generation once v6 join links stop being supported
```

Comment:
Not necessary. This entire class will be removed in the first place one V6 is decommissioned.